### PR TITLE
Fixed an issue where attributes of the same name cannot be expanded when subclasses inherit and hide attributes of the parent class

### PR DESCRIPTION
**Pull request by Dean-ZhenYao-Wang**

### CODE REVIEW

**Comment by Dean-ZhenYao-Wang** [comment]
#644

File: src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs [hunk 1]
```diff
@@ -231,7 +231,11 @@ public virtual Expression CreatePropertyValueExpression(QueryBinderContext conte
             PropertyInfo propertyInfo = source.Type.GetProperty(propertyName, BindingFlags.DeclaredOnly);
             if (propertyInfo == null)
             {
-                propertyInfo = source.Type.GetProperty(propertyName);
+                /*
+                 * History of code:
+                 * propertyInfo = source.Type.GetProperty(propertyName);
+                */
+                propertyInfo = source.Type.GetProperties().Where(m => m.Name.Equals(propertyName)).FirstOrDefault();
```
Comment (xuzhg) [inline src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs:238]: Is it safe to call "FirstOrDefault()" to get the correct property info?

Comment (Dean-ZhenYao-Wang) [reply src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs:238]: After testing, it is found that this modification is not a problem. When the property you are looking for does not exist, the program produces the same result as if it had not been modified.

Comment (Dean-ZhenYao-Wang) [reply src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs:238]: > Is it safe to call "FirstOrDefault()" to get the correct property info?

After testing, this modification was found not to be a problem. When the property you are looking for does not exist, the program produces the same result as if it had not been modified.

Comment (xuzhg) [inline src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs:238]: Would you please add some comments for your change in the source file?

**Review by xuzhg — APPROVED** [review]

File: src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs (hunk 1 above)
Comment (Dean-ZhenYao-Wang) [reply src/Microsoft.AspNetCore.OData/Query/Expressions/SelectExpandBinder.cs:249]: > Would you please add some comments for your change in the source file?

Code comments have been added to the new submission

**Review by xuzhg — APPROVED** [review]

**Pull request merged**